### PR TITLE
Functionnal Copper client

### PR DESCRIPTION
**Pull request by AymericNoel**

### CODE REVIEW

**Comment by AymericNoel** [comment]
I'm on the cleanup of the OpenApi on swaggerHub. This one is not clean enough...

But we have all of the config yet !

File: src/Trakx.Copper.ApiClient.Tests/Secret.cs [hunk 1]
```diff
@@ -18,8 +22,7 @@ static Secrets()
             }
         }
 
-        public static string CopperApiKey => Environment.GetEnvironmentVariable("CopperApiConfiguration__ApiKey")!;
-        public static string CopperApiSecret => Environment.GetEnvironmentVariable("CopperApiConfiguration__ApiSecret")!;
+        public static string? ExchangeApiKey => Environment.GetEnvironmentVariable("CopperApiConfiguration__ApiKey");
```
Comment (monsieurleberre) [inline src/Trakx.Copper.ApiClient.Tests/Secret.cs]: On these new projects, let's make sure we try not to accumulate warnings like this one (you can use `!`)

**Review by monsieurleberre — DISMISSED** [review]
Awesome
Booom 💥💥
If you can just fix Codacy we are good!

**Comment by AymericNoel** [comment]
Codacy told that we should remove partial from classes but it don"t see that these classes are also in ApiClient so the partial isn't gratuitous ... no?

File: src/Trakx.Copper.ApiClient.Tests/Integration/AccountsClientTests.cs [hunk 2]
```diff
@@ -1,52 +1,25 @@
...
-            var account = (await _accountsClient.GetAccountAsync(first.Id)).Result;
-
-            account.Id.Should().Be(first.Id);
-            account.Exchange.Should().Be(first.Exchange);
-            account.IsRebalancing.Should().Be(first.IsRebalancing);
-        }
-
-        [Fact]
-        public async Task GetBalance_should_work()
+        [Fact(Skip = "should have api keys and accounts")]
```
Comment (monsieurleberre) [inline src/Trakx.Copper.ApiClient.Tests/Integration/AccountsClientTests.cs:18]: haha I got some yesterday!

File: src/Trakx.Copper.ApiClient/ClientConfigurator.cs [hunk 3]
```diff
@@ -21,9 +21,6 @@ public ICredentialsProvider GetCredentialProvider(Type clientType)
         {
             switch (clientType.Name)
```
Comment (monsieurleberre) [inline src/Trakx.Copper.ApiClient/ClientConfigurator.cs:22]: let's remove that switch maybe?

**Review by monsieurleberre — CHANGES_REQUESTED** [review]
Excellent! Thanks 🦾🦾💥

**Review by monsieurleberre — APPROVED** [review]

**Pull request merged**